pkg/components/velero/component_test.go

```diff
@@ -195,3 +196,49 @@ component "velero" {
 		t.Fatalf("Rendered manifests shouldn't be empty")
 	}
 }
+
+func TestRenderManifestResticToleration(t *testing.T) {
+	configHCL := `
+component "velero" {
+  provider = "restic"
+  restic {
+    credentials = "foo"
+
+    backup_storage_location {
+      bucket   = "foo"
+      provider = "aws"
+    }
+		
+    tolerations {
+      key = "TestResticToletrationKey"
+      value = "TestResticToletrationValue"
+      operator = "Equal"
+      effect = "NoSchedule"
+      toleration_seconds = "1"
+    }
+  }
+}
+`
+
+	component := NewConfig()
+
+	body, d := util.GetComponentBody(configHCL, Name)
+	if d != nil {
+		t.Fatalf("Error getting component body: %v", d)
+	}
+
+	d = component.LoadConfig(body, &hcl.EvalContext{})
+	if d.HasErrors() {
+		t.Fatalf("Valid config should not return error, got: %s", d)
+	}
+
+	m, err := component.RenderManifests()
+	if err != nil {
+		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
+	}
+
+	renderedDaemonSet := m["velero/templates/restic-daemonset.yaml"]
+	if !strings.Contains(renderedDaemonSet, "key: TestResticToletrationKey") {
+		t.Fatalf("Rendered manifests should have toleration set.")
+	}
```

Comment:
I suggest we use https://github.com/kinvolk/lokomotive/blob/efe364ea57e4d70e063ef6dbbeb304dc5a94f717/pkg/components/contour/component_test.go#L123-L126, which should be more accurate.